client/index.js

```diff
@@ -172,7 +188,9 @@ async function doRender ({ Component, props, hash, err, emitter: emitterProp = e
 
   // We need to clear any existing runtime error messages
   ReactDOM.unmountComponentAtNode(errorContainer)
-  renderReactElement(createElement(App, appProps), appContainer)
+  renderReactElement(<AppContainer errorReporter={ErrorDebugComponent} warnings={false}>
```

Comment:
Why do we move this to here from the `app.js`.? I like this approach. But just need to know why?

And is it possible to use `AppContainer` it only in development?

Reply:
`<AppContainer>` becomes an empty component in production, so there's no harm in including it. I've moved it here since our `componentDidCatch` in `app.js` has to be executed. Also this makes sure we keep the state of app.js (since we expose it to users now)

Reply:
Actually we have to do the webpack magic since in the webpack config we remove react-hot-loader completely (which is currently breaking the tests)